src/backend/plaf.rs

```diff
@@ -0,0 +1,474 @@
+            }
+            cPolyExpr::Mul(es) => {
+                let mut iter = es.iter();
+                let first = self.convert_plaf_poly(iter.next().unwrap());
+                iter.fold(first, |acc, e| acc * self.convert_plaf_poly(e))
+            }
+            cPolyExpr::Neg(e) => -self.convert_plaf_poly(e),
+            cPolyExpr::Pow(e, n) => {
+                if *n == 0 {
+                    pExpr::Const(BigUint::from(1u32))
+                } else {
+                    let e = self.convert_plaf_poly(e);
+                    (1..*n).fold(e.clone(), |acc, _| acc * e.clone())
+                }
+            }
+            cPolyExpr::Halo2Expr(e) => pExpr::from(e),
+            cPolyExpr::Query(column, rotation, annotation) => {
+                let index = self
+                    .c_column_id_to_p_column_index
+                    .get(&column.uuid())
+                    .unwrap();
+                if self.debug {
+                    println!(
+                        "GET c column id {} match p column index {}",
+                        column.uuid(),
+                        index
+                    );
+                    println!("MAP {:#?}", self.c_column_id_to_p_column_index);
+                }
+                pExpr::Var(PlonkVar::Query(
+                    self.convert_plaf_query(column, rotation, annotation, *index),
+                ))
+            }
+        }
+    }
+
+    fn add_id_index_mapping(
+        &self,
+        column: &cColumn,
+        c_column_id_to_p_column_index: &mut HashMap<u32, usize>,
+        counter: &mut usize,
+    ) {
+        c_column_id_to_p_column_index.insert(column.uuid(), *counter);
+        if self.debug {
+            println!(
+                "c column id {} match p column index {}",
+                column.uuid(),
+                counter
+            );
+        }
+        *counter += 1;
+    }
+
+    fn convert_plaf_query(
+        &self,
+        column: &cColumn,
+        rotation: &i32,
+        _annotation: &str,
+        index: usize, // Plaf index starts from 0 for each column type.
+    ) -> ColumnQuery {
+        match column.ctype {
+            cAdvice => ColumnQuery {
+                column: pColumn {
+                    kind: ColumnKind::Witness,
+                    index,
+                },
+                rotation: *rotation,
+            },
+            cFixed => ColumnQuery {
+                column: pColumn {
+                    kind: ColumnKind::Fixed,
+                    index,
+                },
+                rotation: *rotation,
+            },
+            Halo2Advice | Halo2Fixed => {
+                panic!("Imported Halo2Advice and Halo2Fixed are not supported")
+            }
+        }
+    }
+}
+
+// ChiquitoPlafFixedGen currently doesn't work due to reasons mentioned in comments below.
+// We left skeleton code and TODOs for future implementations.
+pub struct ChiquitoPlafFixedGen {
```

Comment:
not a good idea to leave half completed code that does not work, remove ChiquitoPlafFixedGen, just panic if the circuit has fixed_gen or Halo2FixedQuery queriables imported.

Reply:
ok, the panic is already there, remove ChiquitoPlafFixedGen

Reply:
I simply removed all fixed gen related codes. I don't think we should panic if there's fixed gen, because:
1. We already panic when there's imported halo2 fixed column.
2. It's still helpful to output the Plaf struct (without fixed gen) and Plaf Witness struct even if a Chiquito circuit has fixed gen.

Let me know if you disagree though.